source/IJSVGShapeLayer.m

```diff
@@ -17,6 +19,46 @@ - (void)dealloc
     IJSVG_LAYER_DEFAULT_DEALLOC_INSTRUCTIONS
 }
 
-IJSVG_LAYER_ADD_SUBVIEW_DEFAULT_IMPLEMENTATION
+- (void)addSublayer:(CALayer *)layer {
+    if([layer isKindOfClass:[IJSVGLayer class]] == NO &&
+       [layer isKindOfClass:[IJSVGShapeLayer class]] == NO) {
+        NSString * r = [NSString stringWithFormat:@"The layer must be an instance of IJSVGLayer, %@ given.",
+                        [layer class]];
+        NSException * exception = [NSException exceptionWithName:@"IJSVGInvalidSublayerException"
+                                                          reason:r
+                                                        userInfo:nil];
+        @throw exception;
+    }
+    [super addSublayer:layer];
+}
+
+- (void)setBackingScaleFactor:(CGFloat)newFactor
+{
+    if(self.backingScaleFactor == newFactor) {
+        return;
+    }
+    backingScaleFactor = newFactor;
+    self.contentsScale = newFactor;
+    self.rasterizationScale = newFactor;
+    [self setNeedsDisplay];
+};
+
+- (void)renderInContext:(CGContextRef)ctx
+{
+   
+    if(self.blendingMode != kCGBlendModeNormal) {
+        CGContextSaveGState(ctx);
+        CGContextSetBlendMode(ctx, self.blendingMode);
+        [super renderInContext:ctx];
+        CGContextRestoreGState(ctx);
+        return;
+    }
+    [super renderInContext:ctx];
+}
+
+- (CGPoint)absoluteOrigin
+{
+    return CGPointZero;
```

Comment:
I found this by accident but the absolute origin for shapes should always be zero, if we use the calculated value for this particular kind of layer, gradients don't render in the correct location within the mask.